sklearn/model_selection/validate.py

```diff
@@ -0,0 +1,905 @@
+"""
+The :mod:`sklearn.model_selection.validate` module includes classes and
+functions to validate the model.
+"""
+
+# Author: Alexandre Gramfort <alexandre.gramfort@inria.fr>,
+#         Gael Varoquaux <gael.varoquaux@normalesup.org>,
+#         Olivier Girsel <olivier.grisel@ensta.org>
+# License: BSD 3 clause
+
+
+from __future__ import print_function
+from __future__ import division
+
+import warnings
+import numbers
+import time
+
+import numpy as np
+import scipy.sparse as sp
+
+from ..base import is_classifier, clone
+from ..utils import indexable, check_random_state, safe_indexing
+from ..utils.fixes import astype
+from ..utils.validation import _is_arraylike, _num_samples
+from ..externals.joblib import Parallel, delayed, logger
+from ..metrics.scorer import check_scoring
+from .split import check_cv, _safe_split
+
+
+__all__ = ['cross_val_score', 'cross_val_predict', 'permutation_test_score',
+           'learning_curve', 'validation_curve']
+
+
+# TODO Move this into sklearn.exceptions
+class FitFailedWarning(RuntimeWarning):
+    pass
+
+
+def cross_val_score(estimator, X, y=None, labels=None, scoring=None, cv=None,
+                    n_jobs=1, verbose=0, fit_params=None,
+                    pre_dispatch='2*n_jobs'):
+    """Evaluate a score by cross-validation
+
+    Read more in the :ref:`User Guide <validate>`.
+
+    Parameters
+    ----------
+    estimator : estimator object implementing 'fit'
+        The object to use to fit the data.
+
+    X : array-like
+        The data to fit. Can be, for example a list, or an array at least 2d.
+
+    y : array-like, optional, default: None
+        The target variable to try to predict in the case of
+        supervised learning.
+
+    labels : array-like of int with shape (n_samples,), optional
+        Arbitrary domain-specific stratification of the data to be used
+        to draw the splits by the cross validation iterator.
```

Comment:
1) do the labels have to be int?
2) is this docstring copied from somewhere in the codebase? it seems like it wouldn't help me much unless I know exactly what I need already. Again, maybe `labels` should be at the end in the list of arguments, since it's needed only in very specific cases (less common than the need to change `scoring` and `cv`)